[Build stability] Intermittent failure in TestExecuteCheck

```
--- FAIL: TestExecuteCheck (7.85s)
	assertions.go:239: 
                          
	Error Trace:	check_handler_internal_test.go:136
		
	Error:      	Not equal: 
		
	            	expected: 2
		
	            	actual: 0
panic: runtime error: index out of range [recovered]
panic: runtime error: index out of range
```

--- a/agent/check_handler_internal_test.go
+++ b/agent/check_handler_internal_test.go
@@ -2,6 +2,7 @@ package agent
 
 import (
 	"encoding/json"
+	"fmt"
 	"io/ioutil"
 	"os"
 	"path/filepath"
@@ -117,7 +118,7 @@ func TestExecuteCheck(t *testing.T) {
 	metrics := "metric.foo 1 123456789\nmetric.bar 2 987654321"
 	f, err := ioutil.TempFile("", "metric")
 	assert.NoError(err)
-	err = ioutil.WriteFile(f.Name(), []byte(metrics), 0644)
+	_, err = fmt.Fprintln(f, metrics)
 	assert.NoError(err)
 	f.Close()
 	defer os.Remove(f.Name())
@@ -133,7 +134,10 @@ func TestExecuteCheck(t *testing.T) {
 	assert.NoError(json.Unmarshal(msg.Payload, event))
 	assert.NotZero(event.Timestamp)
 	assert.True(event.HasMetrics())
-	assert.Equal(2, len(event.Metrics.Points))
+	if len(event.Metrics.Points) != 2 {
+		fmt.Println(event)
+	}
+	require.Equal(t, 2, len(event.Metrics.Points))
 	metric0 := event.Metrics.Points[0]
 	assert.Equal(float64(1), metric0.Value)
 	assert.Equal("metric.foo", metric0.Name)

--- a/agent/check_handler_internal_test.go
+++ b/agent/check_handler_internal_test.go
@@ -119,7 +119,7 @@ func TestExecuteCheck(t *testing.T) {
 	f, err := ioutil.TempFile("", "metric")
 	assert.NoError(err)
 	_, err = fmt.Fprintln(f, metrics)
-	assert.NoError(err)
+	require.NoError(t, err)
 	f.Close()
 	defer os.Remove(f.Name())
 	checkConfig.OutputMetricFormat = types.GraphiteOutputMetricFormat
@@ -134,10 +134,7 @@ func TestExecuteCheck(t *testing.T) {
 	assert.NoError(json.Unmarshal(msg.Payload, event))
 	assert.NotZero(event.Timestamp)
 	assert.True(event.HasMetrics())
-	if len(event.Metrics.Points) != 2 {
-		fmt.Println(event)
-	}
-	require.Equal(t, 2, len(event.Metrics.Points))
+	require.Equal(t, 2, len(event.Metrics.Points), string(msg.Payload))
 	metric0 := event.Metrics.Points[0]
 	assert.Equal(float64(1), metric0.Value)
 	assert.Equal("metric.foo", metric0.Name)

